Update 70_DENON_AVR.pm Modellabrage im Parse-Bereich eingefügt

Shadow3561 opened the pull request:
Modellabfrage für die Internals.

probiere bitte ob die Ausgabe bei Marantz auch passt. 
Evtl möchtest du es ja auch anders umsetzen, meine Methode ist eher quick and dirty, funktioniert aber.

MfG

Shadow3561 committed:
Update 70_DENON_AVR.pm

Modellabfrage für die Internals.

probiere bitte ob die Ausgabe bei Marantz auch passt. 
Evtl möchtest du es ja auch anders umsetzen, meine Methode ist eher quick and dirty, funktioniert aber.

MfG

Shadow3561 committed:
Update 70_DENON_AVR.pm

Regex für SYMO angepasst auf mir bekannte Denon und Marantz Modelle

delMar43 commented:
Was ist der Vorteil der Model-Abfrage über SYMO, gegenüber den HTTP-Requests?

Shadow3561 commented:
Ich denke so bleibt der Code übersichtlicher.
Ich habe es commited damit du es ins Modul einbauen kannst. Du darfst es
dann auch gerne auskommentieren.

Du schriebst im Forum, dass du das Modul irgendwann so weit bringen
möchtest, dass zwischen alten und neuen Modellen unterschieden wird. Meine
Variante ist ebenso für beide, aber eben ein paar Zeilen kürzer.
Des weiteren belastet es die Telnet verbindung nicht wirklich da das
Internal nur beim Statusrequest und beim define gelesen wird.

Möchte dir ja nur ein wenig Arbeit abnehmen und vielleicht neue Ideen
einbringen.

MfG

Martin Gutenbrunner <notifications@github.com> schrieb am Fr., 6. März
2020, 08:19:

> Was ist der Vorteil der Model-Abfrage über SYMO, gegenüber den
> HTTP-Requests?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/delMar43/FHEM/pull/12?email_source=notifications&email_token=AOUJAWUVYMWGER55HGMVRITRGCPZVA5CNFSM4K72MY32YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEOAKXZY#issuecomment-595635175>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOUJAWX3CX4NK6MBHPDIY4TRGCPZVANCNFSM4K72MY3Q>
> .
>

Shadow3561 commented:
Wäre es nicht möglich das Modul so umzubauen, dass die älteren Modelle per http gesteuert werden und die, die Mehrfachverbindungen per Telnet zulassen über Telnet?
Die Telnet-Befehle können ja auch per http an die älteren AVR gesendet werden.
z.B.
http://{ip}/goform/formiPhoneAppDirect.xml?MVUP

Ich habe leider nur keine Möglichkeit die Befehle oder Statusantworten vom AVR zu überprüfen, da das bei meinem nicht mehr funktioniert.

MfG

delMar43 commented:
Ja, genau das habe ich gemeint,als ich gesagt habe, das Modul soll alle Modelle unterstützen.
Und mit der Erkennung von deviceinfo oder dem ajax request wissen wir auch schon, was für den jeweiligen avr die richtige Variante ist.

Es ist zur Zeit leider etwas eng mit der Freizeit, aber ich verfolge alles mit und werde zeitnah wieder ein Update bringen

Shadow3561 commented:
Ich bin mir nicht wirklich sicher, aber ich denke die AVR, die HEOS integriert haben unterstützen mehrere Telnet-Verbindungen. HEOS CLI kommuniziert ja über Telnet Port 1255. Da man auch mehrere Heos Systeme haben kann, sollte es wohl viele Telnet-Verbindungen haben. Aber auch dies lässt sich nur im Forum prüfen. 
Des weiteren funktionieren zumindest bei meinem AVR die ganzen Readings für die Titel (NSE) und Abspielzeiten nicht. Laut Denon Support werden sie nur noch über die Heos Cli ausgegeben.

MfG

Shadow3561 committed:
Add files via upload

Modul to read Solaredge SMI Devices via ModBus over RS485

Shadow3561 committed:
Add files via upload

new DAB+ readings,
parse Product manufacture changed,
parse product name changed

Shadow3561 commented:
add dab+ readings
change parse device name
change parse model
change parse manufacture
add  some things in the PS section
and much more
